executor/executor_simple.go

```diff
@@ -184,6 +186,60 @@ func (e *SimpleExec) executeCreateUser(s *ast.CreateUserStmt) error {
 	return nil
 }
 
+func (e *SimpleExec) executeAlterUser(s *ast.AlterUserStmt) error {
+	if s.CurrentAuth != nil {
+		user := e.ctx.GetSessionVars().User
+		if len(user) == 0 {
+			return errors.New("Session user is empty")
+		}
+		spec := &ast.UserSpec{
+			User:    user,
+			AuthOpt: s.CurrentAuth,
+		}
+		s.Specs = []*ast.UserSpec{spec}
+	}
+
+	failedUsers := make([]string, 0, len(s.Specs))
+	for _, spec := range s.Specs {
+		userName, host := parseUser(spec.User)
+		exists, err1 := userExists(e.ctx, userName, host)
+		if err1 != nil {
+			return errors.Trace(err1)
+		}
+		if !exists {
+			failedUsers = append(failedUsers, spec.User)
+			if s.IfExists {
+				// TODO: Make this error as a warning.
+			}
```

Comment:
No. Add failed users to `failedUsers` enough here. In the end of the function we will return an error.